backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchStringIndexFieldTypeOptionsStep.java

```diff
@@ -213,7 +213,7 @@ public IndexFieldType<String> toIndexFieldType() {
 			builder.queryElementFactory( PredicateTypeKeys.REGEXP, new ElasticsearchTextRegexpPredicate.Factory() );
 			builder.queryElementFactory( PredicateTypeKeys.TERMS, new ElasticsearchTermsPredicate.Factory<>( codec ) );
 			builder.queryElementFactory( ElasticsearchPredicateTypeKeys.SIMPLE_QUERY_STRING,
-					new ElasticsearchSimpleQueryStringPredicateBuilderFieldState.Factory() );
+					new ElasticsearchCommonQueryStringPredicateBuilderFieldState.Factory() );
```

Comment:
Seems weird that the key is still named `SIMPLE_QUERY_STRING`?

Also, this won't play well with the metamodel: the field type will have a trait named `predicate:simple-query-string` and no trait named `predicate:query-string`.

You probably need two separate keys here.

And you need to look at the metamodel test for this predicate, cause there's something missing or wrong if this didn't get detected :D